From d75356728f52bd298a76a51256ea42841f6eefbd Mon Sep 17 00:00:00 2001
From: Yang Ni <yangni@google.com>
Date: Mon, 13 Jun 2016 11:18:40 -0700
Subject: [PATCH] Disable AVX support in bcc

Bug: 28879581

LLVM generates AVX code for long3 load/store, which treats a long3 value
as 256 bits, while the RS runtime only allocates 192 bits for a long3
element in an allocation.

This CL disables AVX codegen to avoid this issue.

Change-Id: Ia656a36c35a4cfbe620a555abb8b28c808843459
---
 lib/Support/CompilerConfig.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/lib/Support/CompilerConfig.cpp b/lib/Support/CompilerConfig.cpp
index b2b3a55..87fcdad 100644
--- a/lib/Support/CompilerConfig.cpp
+++ b/lib/Support/CompilerConfig.cpp
@@ -238,6 +238,11 @@ bool CompilerConfig::initializeArch() {
     attributes.push_back("+sse4.2");
 #endif
 #endif // __HOST__
+
+    // LLVM generates AVX code that treats a long3 as 256 bits, while
+    // RenderScript considers a long3 192 bits (http://b/28879581)
+    attributes.push_back("-avx");
+
     break;
 #endif  // PROVIDE_X86_CODEGEN

@@ -265,6 +270,10 @@ bool CompilerConfig::initializeArch() {
       attributes.push_back("+f16c");
 #endif // __HOST__

+    // LLVM generates AVX code that treats a long3 as 256 bits, while
+    // RenderScript considers a long3 192 bits (http://b/28879581)
+    attributes.push_back("-avx");
+
     break;
 #endif  // PROVIDE_X86_CODEGEN

--
2.7.4
